src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java

```diff
@@ -296,9 +307,9 @@ public void execute() {
             log("compiled on " + compileTimestamp, Project.MSG_VERBOSE);
 
             // Check for no arguments
-            if (fileName == null && fileSets.isEmpty()) {
+            if (fileName == null && fileSets.isEmpty() && sourcePaths.isEmpty()) {
```

Comment:
>  Is this how all out future conditions should be?

I think that it should. When I started my contribution to Checkstyle project @romani usually asked to do this.  Usually it really helps to see where there is a missing branch in the coverage report. It is easy to decide what should be covered if there are two branch in the condition, however when it comes to more than two branches...